File: Python_WaltisExamples/Code_14_fHoch3/F_hoch_3/cgi-bin/getAlertDetails.py
#!/bin/python

# START------------------------------------------------------------------------
# Author:        Walter Rothlin
# Description:   getAlertDetails
#
#
# Calling:       getAlertDetails.py
# REST service to get more details about a mailing adress pattern (Peterliwiese 33 8855 --> Walter Rothlin Peterliwiese 33 8855 Wangen SZ +41 55 460 14 40 47 47.1723 / 8.8674
# History:
# 04-Mar-2020    V1.0   Walter Rothlin	Initial Version
#
# END--------------------------------------------------------------------------

import requests 

print("HTTP/1.1 200 OK")
print("Access-Control-Allow-Origin: *")
print("Content-type: application/json\n")

# api-endpoint 
URL = "http://maps.googleapis.com/maps/api/geocode/json"
URL = "https://maps.googleapis.com/maps/api/place/findplacefromtext/json?input=${address}&inputtype=textquery&fields=photos,formatted_address,name,rating,opening_hours,geometry&key=AIzaSyDAKgrjrKNmBPu47TUFP-x8hY_jp2Ainbk"
 
# location given here 
location = "delhi technological university"
  
# defining a params dict for the parameters to be sent to the API 
# PARAMS = {'address':location} 
  
# sending get request and saving the response as response object 
r = requests.get(url = URL, params = PARAMS) 
  
# extracting data in json format 
data = r.json() 
  
  
# extracting latitude, longitude and formatted address  
# of the first matching location 
latitude = data['results'][0]['geometry']['location']['lat'] 
longitude = data['results'][0]['geometry']['location']['lng'] 
formatted_address = data['results'][0]['formatted_address'] 
  
# printing the output 
print("Latitude:%s\nLongitude:%s\nFormatted Address:%s"
      %(latitude, longitude,formatted_address)) 

File: Python_WaltisExamples/Code_14_fHoch3/F_hoch_3/cgi-bin/getAlertDetails.pl
#!/usr/bin/perl

# START------------------------------------------------------------------------
# Author:        Walter Rothlin
# Description:   getAlertDetails
#
#
# Calling:       getAlertDetails.py
# REST service to get more details about a mailing adress pattern (Peterliwiese 33 8855 --> Walter Rothlin Peterliwiese 33 8855 Wangen SZ +41 55 460 14 40 47 47.1723 / 8.8674
# History:
# 04-Mar-2020    V1.0   Walter Rothlin	Initial Version
#
# END--------------------------------------------------------------------------
use CGI qw(param);
use lib "$ENV{DOCUMENT_ROOT}/cgi-bin";
use littlePerlLib;
use LWP::Simple;

print("HTTP/1.1 200 OK\n");
print("Access-Control-Allow-Origin: *\n");
print("Content-type: application/json\n\n");

$searchPattern  = getParam("searchPattern","Peterliwiese+33+8855+Wangen");

$searchXMLResponse = getHttpSimple("https://tel.search.ch/api/?wo=".${searchPattern}."&key=8e8a84fd0f10d3b44920e49bc3b06a37");

$tel_name      = getTagValueFromXML("tel:name",      $searchXMLResponse);
$tel_firstname = getTagValueFromXML("tel:firstname", $searchXMLResponse);
$tel_street    = getTagValueFromXML("tel:street",    $searchXMLResponse);
$tel_streetno  = getTagValueFromXML("tel:streetno",  $searchXMLResponse);
$tel_zip       = getTagValueFromXML("tel:zip",       $searchXMLResponse);
$tel_city      = getTagValueFromXML("tel:city",      $searchXMLResponse);
$tel_canton    = getTagValueFromXML("tel:canton",    $searchXMLResponse);
$tel_country   = getTagValueFromXML("tel:country",   $searchXMLResponse);
$tel_phone     = getTagValueFromXML("tel:phone",     $searchXMLResponse);
$tel_extra     = getTagValueFromXML("tel:extra ",    $searchXMLResponse);

## print($tel_name);
print("{\n");
print("   \"Name\"     :\"".$tel_name."\",\n");
print("   \"FirstName\":\"".$tel_firstname."\",\n");
print("   \"Street\"   :\"".$tel_street."\",\n");
print("   \"No\"       :\"".$tel_streetno."\",\n");
print("   \"Zip\"      :\"".$tel_zip."\",\n");
print("   \"City\"     :\"".$tel_city."\",\n");
print("   \"Canton\"   :\"".$tel_canton."\",\n");
print("   \"Country\"  :\"".$tel_country."\",\n");
print("   \"Phone\"    :\"".$tel_phone."\",\n");
print("   \"Mobile\"   :\"".$tel_extra."\"\n");

print("}\n");